sycl/source/detail/memory_manager.cpp

```diff
@@ -47,6 +47,8 @@ void MemoryManager::release(ContextImplPtr TargetContext, SYCLMemObjI *MemObj,
 
 void MemoryManager::releaseImageBuffer(ContextImplPtr TargetContext,
                                        void *ImageBuf) {
+  (void)TargetContext;
```

Comment:
late nit: suggest to introduce macros like
`#define UNUSED_ARGUMENT(x) (void)x`
`#define UNUSED_ARGUMENTS2(x,y) (void)x; (void) y`
 and them instead.

Reply:
Please not in this patch as we already have hundreds of places using this old style "(void)TargetContext".
It should be a separate PR adding those macros, imo.

Reply:
Separate patch sounds good to me.